src/transformers/trainer.py

```diff
@@ -2222,9 +2229,11 @@ def _load_best_model(self):
                                     "using `TrainerCallback` to save adapter_model.bin in corresponding folders, "
                                     "here are some examples https://github.com/huggingface/peft/issues/96"
                                 )
+                                has_been_loaded = False
                         else:
                             # We can't do pure 8bit training using transformers.
                             logger.warning("Could not loading a quantized checkpoint.")
+                            has_been_loaded = False
```

Comment:
should this be removed now?

Reply:
I think this is needed so that it can be used in the block below for the check, otherwise it will throw an error similar as #24096

Reply:
proposed something in bf31c5e